tv_apps/tv_shared/test/unit/mocks/mock_presentation_session.js

```diff
@@ -0,0 +1,92 @@
+(function(exports) {
+  'use strict';
+
+  // Mock API from http://bit.ly/1uAh0jd
+  var MockPresentationSession = {
+    _id: undefined,
+
+    get id () {
+      return this._id;
+    },
+
+    _mEventHandlers: {
+      'statechange': [],
+      'message': []
+    },
+
+    _onstatechange: undefined,
+
+    set onstatechange (handler) {
+      if (typeof handler === 'function') {
+        this._onstatechange = handler;
+      }
+    },
+
+    _onmessage: undefined,
+
+    set onmessage (handler) {
+      if (typeof handler === 'function') {
+        this._onmessage = handler;
+      }
+    },
+
+    send: function(data) {
+    },
+
+    disconnect: function() {
+      this._state = 'disconnected';
+    },
+
+    // two states: 'connected' and 'disconnected'
+    _state: 'disconnected',
+
+    get state () {
+      return this._state;
+    },
+
+    addEventListener: function(eventName, handler) {
```

Comment:
Since there are duplicated codes for sending/receiving events.. did you tried `mock_event_target.js` under `shared`?

It's ok for me to leave these code as is though.